### PR TITLE
added few touches on firebase and data models

### PR DESCRIPTION
**Pull request by aalmsaodi**
We probably should add a singleton (shared instance) for the list of acts. We should import the list from FireBase once the app get launched.

### CODE REVIEW

File: Everest/Everest/Models/Act.swift [hunk 1]
```diff
@@ -13,22 +13,16 @@ import SwiftyJSON
 class Act: NSObject {
 
   var id: String
-  var category: String
```
Comment (kavitagaitonde) [inline Everest/Everest/Models/Act.swift:16]: We should leave the category in the Act.

File: Everest/Everest/Models/Action.swift [hunk 2]
```diff
@@ -13,23 +13,20 @@ import SwiftyJSON
 class Action: NSObject {
   
   var id: String          //same id from acts
-  var categoryId: String //same categoryId Id from acts
   var createdAt: String
```
Comment (kavitagaitonde) [inline Everest/Everest/Models/Action.swift:16]: Why is this removed?

File: Everest/Everest/Models/User.swift [hunk 3]
```diff
@@ -45,9 +45,9 @@ class User: NSObject {
     if let momentIds = user["momentIds"].dictionary {
       self.momentIds = momentIds.map { $1.string! }
     }
-    if let ideas = user["ideas"].array {
-      self.ideas = ideas.map { Act(act: $0) }
-    }
+//    if let ideas = user["ideas"].array {
+//      self.ideas = ideas.map { Act(id: String, title: <#String#>) }
+//    }
```
Comment (kavitagaitonde) [inline Everest/Everest/Models/User.swift:50]: do we need this code? whats an idea?

File: Everest/Everest/Models/Act.swift (hunk 1 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/Act.swift:16]: I couldn't find a working way to convert, csv file to json with the previous additional info. That's why I dropped the additional info. However, we should still get all desired functionality with slight work on the app side.

File: Everest/Everest/Models/Action.swift (hunk 2 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/Action.swift:16]: Categories won't have an id, their title will server as an identifier as well (since they are limited in number and their titles are small in size). This reduction was needed for auto filling for the actPicker list through json import. We can add it later if needed through manual edit on FB.

File: Everest/Everest/Models/User.swift (hunk 3 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/User.swift:50]: This code is to remind us to add a functionality for user to suggest new acts.

File: Everest/Everest/Models/Action.swift (hunk 2 above)
Comment (kavitagaitonde) [reply Everest/Everest/Models/Action.swift:16]: category will be a string like "surprise". its more like a type not id.

File: Everest/Everest/Models/User.swift (hunk 3 above)
Comment (kavitagaitonde) [reply Everest/Everest/Models/User.swift:50]: ok then we should refer to it as a userAct so its relatable.

Comment (aalmsaodi) [reply Everest/Everest/Models/User.swift:50]: sure

File: Everest/Everest/Models/Action.swift (hunk 2 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/Action.swift:16]: I see what you see, but it can work as an id as well for the reason mentioned above.

Comment (kavitagaitonde) [reply Everest/Everest/Models/Action.swift:16]: im ok with id OR type. my point is that there should be a reference to the type of category from a given act.
we will also need this when we support user created acts.

Comment (kavitagaitonde) [reply Everest/Everest/Models/Action.swift:16]: also we need to add a field like "type" whose value can be "system or user"

Comment (aalmsaodi) [reply Everest/Everest/Models/Action.swift:16]: Sure you can add the a reference to category in action, but this info will be included in the actId Prefix.
I don't think we should have a user type anymore. User can suggest but should not be allowed to add acts, because we will have a lot of repetition in acts. Rather, the user will be able to suggest new acts (ideas), and then it is up to us to add them if they are not already existing in different wording.

File: Everest/Everest/Models/Act.swift (hunk 1 above)
Comment (kavitagaitonde) [reply Everest/Everest/Models/Act.swift:16]: There is a way to convert excel file to json format which we can push to the FB db.

File: Everest/Everest/Models/Action.swift (hunk 2 above)
Comment (kavitagaitonde) [reply Everest/Everest/Models/Action.swift:16]: We can disallow users from adding via the app. But when we do support adding it will be good to have the context of which system created it.

File: Everest/Everest/Models/Act.swift (hunk 1 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/Act.swift:16]: I already tried with excel format and csv format. I couldn't produce the desired results with the excel file.
If you want to to give it a shot, please feel free. If you succeed, feel free to change the database as you see needed.

File: Everest/Everest/Models/Action.swift (hunk 2 above)
Comment (aalmsaodi) [reply Everest/Everest/Models/Action.swift:16]: It doesn't harm to add it if you want.
